Add try-except statements

MarisZul opened the pull request:
Function ic_filter is altered with two try-except statements to give out more meaningful error-message if max_distance is set too small. Related to issue #54.

MarisZul committed:
Add try-except statements

Function ic_filter is altered with two try-except statements to give out more meaningful error-message if max_distance is set too small. Related to issue #54.

cchwala commented on src/pypwsqc/indicator_correlation.py:
```diff
@@ -208,13 +208,22 @@ def ic_filter(
+    except ValueError as e:
+        if "([0])" in str(e):
+           print("Insert a valid bin_size")
+           return
```
Should the user be informed here that the ic_filter fails? If yes, then raise again a `ValueError`, but with the meaningful error message that you have in the print statement.

You should not cat an exception and then silently return None because this might results in unexpected behavior of the code that is calling `ic_filter`.

The same applies to the except statements below.

cchwala commented on src/pypwsqc/indicator_correlation.py:
```diff
@@ -229,25 +238,30 @@ def rsw(m):
-        pws_indcorr_score_list.append(score)
+    try:
+        for pws_id in indicator_correlation_matrix["id_neighbor"].values:  # noqa: PD011
+            binned_indcorr_pws = (
```
Here, I would keep the try-except statement as short as possible. As far as I understand you only want to catch errors from `groupby_bins` and not frmo any of the lines below that are within the try block.

MarisZul committed:
Reraise ValueError

In the try-except statements the ValueError is now reraised with a more meaningful error message. Issue #54

cchwala commented on src/pypwsqc/indicator_correlation.py:
```diff
@@ -208,13 +208,22 @@ def ic_filter(
+    except ValueError as e:
+        if "([0])" in str(e):
+           print("Insert a valid bin_size")
+           raise
```
use `raise ValueError("Insert a valid bin_size") to have a specific error message for the user.

this also applies below

cchwala commented on src/pypwsqc/indicator_correlation.py:
```diff
@@ -230,11 +239,16 @@ def rsw(m):
+                .quantile(quantile_bin_pws, skipna=True)
+            )
+        except:
+            ValueError
```
this has to be move to the line above

cchwala commented:
When adding a message to an error that is raised, it is recommended to do it the following way

```python
msg = 'Some error message'
raise ValueError(msg)
```

At least, this is what ruff will recommend and complain about... 😉 see [here](https://docs.astral.sh/ruff/rules/raw-string-in-exception/)

MarisZul committed:
Raise better Error msg

Edit try-except and error message without hiding the original traceback while being ruff compliant

MarisZul committed:
Add new tests

Additional tests are added to cover the newly added lines with the clearer error message. Related to issue #54

MarisZul committed:
Fix tests

Alter the string of the error msg and ignore the RuntimeWarning for the last test. #54

MarisZul committed:
Correct Regex

AssertionError, because of false string separation over two lines. #54

MarisZul committed:
Set max_dist in test

to 1200 for consistency reasons so that its the same when applying the ic_filter and calculating the matrices. #54
